pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,26 @@
+package agent
+
+// BareMetalPlatform defines agent based install configuration specific to bare metal clusters.
+// Can only be used with spec.installStrategy.agent.
+type BareMetalPlatform struct {
+	// APIVIP is the virtual IP used to reach the OpenShift cluster's API.
+	APIVIP string `json:"apiVIP"`
+
+	// APIVIPDNSName is the domain name used to reach the OpenShift cluster API.
+	APIVIPDNSName string `json:"apiVIPDNSName"`
```

Comment:
Why is that required? in assisted we are setting it only for day2 clusters.

Reply:
Updated to optional.